File: src/main/java/com/study/platform/domain/team/model/TeamSchedule.java
package com.study.platform.domain.team.model;

import com.study.platform.global.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "team_schedules")
public class TeamSchedule extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private StudyTeam team;

    @Column(length = 100, nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime scheduledAt;

    @Builder
    private TeamSchedule(StudyTeam team, String title, String description, LocalDateTime scheduledAt) {
        this.team = team;
        this.title = title;
        this.description = description;
        this.scheduledAt = scheduledAt;
    }

    public static TeamSchedule create(StudyTeam team, String title, String description, LocalDateTime scheduledAt) {
        return TeamSchedule.builder()
                .team(team)
                .title(title)
                .description(description)
                .scheduledAt(scheduledAt)
                .build();
    }

    public void update(String title, String description, LocalDateTime scheduledAt) {
        this.title = title;
        this.description = description;
        this.scheduledAt = scheduledAt;
    }
}

File: domain/src/main/kotlin/com/study/platform/domain/team/model/TeamSchedule.kt
package com.study.platform.domain.team.model

import com.study.platform.global.entity.BaseTimeEntity
import com.study.platform.global.exception.CustomException
import com.study.platform.global.exception.ErrorCode
import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "team_schedules")
class TeamSchedule : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "BINARY(16)")
    var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    var team: StudyTeam? = null

    @Column(length = 100, nullable = false)
    var title: String = ""

    @Column(columnDefinition = "TEXT")
    var description: String? = null

    @Column(nullable = false)
    var scheduledAt: LocalDateTime? = null

    fun update(title: String, description: String?, scheduledAt: LocalDateTime) {
        validateTitle(title)
        validateDescription(description)
        this.title = title
        this.description = description
        this.scheduledAt = scheduledAt
    }

    fun validateBelongsToTeam(teamId: UUID) {
        if (team?.id != teamId) throw CustomException(ErrorCode.TEAM_SCHEDULE_NOT_FOUND)
    }

    companion object {
        private const val MAX_TITLE_LENGTH = 100
        private const val MAX_DESCRIPTION_LENGTH = 1000

        fun create(team: StudyTeam, title: String, description: String?, scheduledAt: LocalDateTime): TeamSchedule {
            validateTitle(title)
            validateDescription(description)
            return TeamSchedule().also {
                it.team = team
                it.title = title
                it.description = description
                it.scheduledAt = scheduledAt
            }
        }

        private fun validateTitle(title: String) {
            if (title.isBlank()) throw CustomException(ErrorCode.INVALID_INPUT)
            if (title.length > MAX_TITLE_LENGTH) throw CustomException(ErrorCode.INVALID_INPUT)
        }

        private fun validateDescription(description: String?) {
            if (description != null && description.length > MAX_DESCRIPTION_LENGTH) throw CustomException(ErrorCode.INVALID_INPUT)
        }
    }
}
